crypto/src/key/signature/mod.rs

```diff
@@ -114,9 +128,45 @@ mod test {
         let msg = b"abc";
 
         // we signed the message above and stored the encoded data. Now it has to work from decoded data
-        let sig_hex = "000101aa39cdbb96b4eec724cac0400a30cf0d2b9a1040d3aa4f58a34a218d03349f730b8dc2e9b592c6eaac524bc7a5266815f47633aa6eb58708ee262667629a1b86";
-        let pk_hex = "0080283462ee4f0840e21d6de7744ba42929d1b74b7a948e8229d9551e7760ec8c52";
-        let sk_hex = "000101181b259bac04d8ec3f6ea2a86b37f39a353288a8410fc469b9f2d5c59ce30a36c10bfdc906c8343fe0fb42c2564d6b1d3bf8ae3d73f0f7e5424cb60a9639d7e0";
+        let sig_hex = "003c002dd5ea8f05240394eb109b8e9b52716db2720cebbc5dd66394afdf761d2ef76dc8a292d3a44e4d9f1f8fd8e17dc404e317082f5a2ce8adafde01e766aaa6";
+        let pk_hex = "00dff1d77f2a671c5f36183726db2341be58feae1da2deced843240f7b502ba659";
+        let sk_hex = "00b7e151628aed2a6abf7158809cf4f3c762e7160f38b4da56a784d9045190cfef";
+
+        let sig_bin: Vec<u8> = FromHex::from_hex(sig_hex).unwrap();
+        let pk_bin: Vec<u8> = FromHex::from_hex(pk_hex).unwrap();
+        let sk_bin: Vec<u8> = FromHex::from_hex(sk_hex).unwrap();
+
+        let pk = PublicKey::decode(&mut pk_bin.as_slice()).unwrap();
+        let sk = PrivateKey::decode(&mut sk_bin.as_slice()).unwrap();
+        let sig = Signature::decode(&mut sig_bin.as_slice()).unwrap();
+
+        assert!(pk.verify_message(&sig, msg));
+        assert_eq!(PublicKey::from_private_key(&sk), pk);
+    }
+
+    #[rstest]
+    #[trace]
+    #[case(Seed::from_entropy())]
+    fn serialize2(#[case] seed: Seed) {
+        let mut rng = make_seedable_rng(seed);
+        let (sk, pk) = PrivateKey::new_from_rng(&mut rng, KeyKind::RistrettoSchnorr);
+        let msg = b"abc";
+        let sig = sk.sign_message(msg).unwrap();
+        assert!(pk.verify_message(&sig, msg));
+
+        let encoded_sig = sig.encode();
+        let decoded_sig = Signature::decode(&mut encoded_sig.as_slice()).unwrap();
+        assert_eq!(decoded_sig, sig);
+    }
+
+    #[test]
+    fn serialize_chosen_data2() {
```

Comment:
I guess here instead of 2 (and no number in the other instance) you can put the names of the algos because we don't have to read the first byte every time :smile: